### ISSUE TITLE
Snapcast server starts, no Spotify connect device appears though

### ISSUE THREAD

**Issue by febs** [djmaze/librespot-snapserver#8]
Using ubuntu server on a raspberry Pi 3B+.

Using **mazzolino/librespot-snapserver:arm** as per the other issue. As expected and thescribed in #7 , the default image won't run.

No spotify connect device is added to the list, though. Other spotify connect devices (using raspotify) are showing up correctly, for what it matters.

If ran without detaching the terminal, here's what I get. I see no line related to librespot.
`ubuntu@ubuntu:~$ sudo docker run --name snapserver --net host --rm -e DEVICE_NAME=Snapcast mazzolino/librespot-snapserver:arm `

`2021-01-06 11-08-29.830 [Info] (main) Adding source: pipe:///tmp/snapfifo?name=default`
`2021-01-06 11-08-29.831 [Notice] (init) Settings file: "/root/.config/snapserver/server.json"`
`2021-01-06 11-08-29.833 [Error] (Avahi) Failed to create client: Daemon not running`
`2021-01-06 11-08-29.834 [Info] (PcmStream) PcmStream: default, sampleFormat: 48000:16:2`
`2021-01-06 11-08-29.834 [Info] (PcmStream) Stream: default, metadata={`
`2021-01-06 11-08-29.834 [Info] (PcmStream)     "STREAM": "default"`
`2021-01-06 11-08-29.834 [Info] (PcmStream) }`
`2021-01-06 11-08-29.834 [Info] (Server) onMetaChanged (default)`
`2021-01-06 11-08-29.835 [Info] (PipeStream) PipeStream mode: create`
`2021-01-06 11-08-29.835 [Info] (Server) Stream: {"fragment":"","host":"","path":"/tmp/snapfifo","query":{"chunk_ms":"20","codec":"flac","name":"default","sampleformat":"48000:16:2"},"raw":"pipe:////tmp/snapfifo?chunk_ms=20&codec=flac&name=default&sampleformat=48000:16:2","scheme":"pipe"}`
`2021-01-06 11-08-29.837 [Error] (PosixStream) Exception: end of file`
`2021-01-06 11-08-29.838 [Info] (ControlServer) Creating TCP acceptor for address: 0.0.0.0, port: 1705`
`2021-01-06 11-08-29.838 [Info] (ControlServer) Creating HTTP acceptor for address: 0.0.0.0, port: 1780`
`2021-01-06 11-08-29.838 [Info] (StreamServer) Creating stream acceptor for address: 0.0.0.0, port: 1704`
`2021-01-06 11-08-29.839 [Info] (main) number of threads: 4, hw threads: 4`
`2021-01-06 11-08-30.242 [Notice] (StreamServer) StreamServer::NewConnection: 192.168.8.129`
`2021-01-06 11-08-30.247 [Info] (Server) Hello from 15133707-0ad4-46b4-862c-1d9a7ce7cfa1, host: SM-A405FN, v0.22.0, ClientName: Snapclient, OS: Android 10, Arch: arm64-v8a, Protocol version: 2`

How can I help debug this? 
Thanks,

**Comment by f-necas** [comment]
Hi @febs,

I had the same problem with amd64 tag. I've used latest-amd64 and it works well. Did you try this ?

**Comment by febs** [comment]
Thank you very much for your suggestion @f-necas . I ended up dropping this solution altogether and building the components by hand the old school way.

**Comment by Jakobovski** [comment]
> Hi @febs,
> 
> I had the same problem with amd64 tag. I've used latest-amd64 and it works well. Did you try this ?

Same issue here

**Comment by djmaze** [comment]
I just updated the image to snapcast 0.25.0 and librespot 0.2.0. Everything built successfully and it seems to work on all architectures. Please try the latest version. (Or the `0.25.0` tag which will soon be pushed as well.)

**Comment by Drun555** [comment]
> I just updated the image to snapcast 0.25.0 and librespot 0.2.0. Everything built successfully and it seems to work on all architectures. Please try the latest version. (Or the `0.25.0` tag which will soon be pushed as well.)

Unfortunately, right now it's not working once again. Logs is exactly the same as above.

**Comment by Drun555** [comment]
Just tried once again after an hour, and now it's appearing... Strange

**Comment by djmaze** [comment]
Well, no changes to the image in the meantime, so it should work all the time..

**Comment by Drun555** [comment]
Yes, I have no problems for already a week. Thank you for your work!

**Comment by djmaze** [comment]
Anyone still having problems? Otherwise I will close this.

**Comment by Drun555** [comment]
I haven't had any issues since then, but I would be very happy if you could consider isolating the snapserver.conf from the container. I saw someone already mention such a request - it would be great if we could use snapserver's built-in modularity with this container (for example, I would really like to connect shairport)

**Comment by schulzh** [comment]
I am facing the same issue. It was working just fine before, but today Snapcast does not show up in Spotify connect anymore on my smartphone. Strangely, it does show up on my desktop computer and I can start it there, then control it with my smartphone.
Recreating the docker container did not help. Logs:
```
server_1             | 2023-02-11 17-49-23.100 [Info] (Snapserver) Version 0.26.0
server_1             | 2023-02-11 17-49-23.101 [Info] (Snapserver) Adding source: librespot:///librespot?name=Spotify&devicename=Snapcast&bitrate=320&volume=100
server_1             | 2023-02-11 17-49-23.103 [Notice] (init) Settings file: "/root/.config/snapserver/server.json"
server_1             | 2023-02-11 17-49-23.103 [Info] (Snapserver) Using HTTP host name: server
server_1             | 2023-02-11 17-49-23.104 [Error] (Avahi) Failed to create client: Daemon not running
server_1             | 2023-02-11 17-49-23.104 [Info] (PcmStream) PcmStream: Spotify, sampleFormat: 44100:16:2
server_1             | 2023-02-11 17-49-23.105 [Info] (Server) Stream: {"fragment":"","host":"","path":"/librespot","query":{"bitrate":"320","chunk_ms":"20","codec":"flac","devicename":"Snapcast","name":"Spotify","sampleformat":"44100:16:2","volume":"100"},"raw":"librespot:////librespot?bitrate=320&chunk_ms=20&codec=flac&devicename=Snapcast&name=Spotify&sampleformat=44100:16:2&volume=100","scheme":"librespot"}
server_1             | 2023-02-11 17-49-23.105 [Info] (FlacEnc) Init - compression level: 2
server_1             | 2023-02-11 17-49-23.107 [Info] (Watchdog) Starting watchdog, timeout: 7800s
server_1             | 2023-02-11 17-49-23.107 [Info] (ControlServer) Creating TCP acceptor for address: 0.0.0.0, port: 1705
server_1             | 2023-02-11 17-49-23.107 [Info] (ControlServer) Creating HTTP acceptor for address: 0.0.0.0, port: 1780
server_1             | 2023-02-11 17-49-23.107 [Info] (StreamServer) Creating stream acceptor for address: 0.0.0.0, port: 1704
server_1             | 2023-02-11 17-49-23.107 [Info] (Snapserver) Number of threads: 4, hw threads: 4
server_1             | 2023-02-11 17-49-23.112 [Info] (librespot) librespot 0.4.2 UNKNOWN (Built on 2022-10-02, Build ID: NwjPfcVW, Profile: release)
server_1             | 2023-02-11 17-49-23.374 [Notice] (StreamServer) StreamServer::NewConnection: 172.20.0.2
server_1             | 2023-02-11 17-49-23.378 [Info] (Server) Hello from 02:42:ac:14:00:02, host: cd4fd4ce8c42, v0.26.0, ClientName: Snapclient, OS: Alpine Linux v3.16, Arch: x86_64, Protocol version: 2
server_1             | 2023-02-11 17-49-23.491 [Notice] (StreamServer) StreamServer::NewConnection: 172.20.0.3
server_1             | 2023-02-11 17-49-23.497 [Info] (Server) Hello from 02:42:ac:14:00:03, host: e0060d5ce499, v0.26.0, ClientName: Snapclient, OS: Alpine Linux v3.16, Arch: x86_64, Protocol version: 2
```
